AdvancedHero.cpp

```diff
@@ -9,7 +10,33 @@ AdvancedHero::AdvancedHero(const std::string& name, int hp, const int dmg, const
 
 AdvancedHero AdvancedHero::parseUnit(const std::string& filename){
     Hero Unit = Hero::parseUnit(filename);
-    return AdvancedHero(Unit.getName(), Unit.getHP(), Unit.getDmg(), Unit.getAttackSpeed());
+    return AdvancedHero(Unit.getName(), Unit.getHP(), Unit.getDmg(), Unit.getAttackCooldown());
+}
+
+void AdvancedHero::advancedBattle(AdvancedHero* target){
+    double w1 = this->getAttackCooldown();
+    double w2 = target->getAttackCooldown();
+    double tempw1 = 0;
+    double tempw2 = 0;
+    
+    this->advancedDamage(target);
+    
+    while(!this->isDead() && !target->isDead()){
+        
+        w1 = this->getAttackCooldown();
+        w2 = target->getAttackCooldown();
+        
+        if(w1+tempw1 < w2+tempw2){
```

Comment:
Na itt már jó. Csak ennek nem kellett volna egy külön `AdvancedHero::advancedBattle` függvényt írni, hanem a `Hero::Battle` függvényét kellett volna kijavítani. A `damaging`nek virtuálisnak kéne lennie, és akkor azt lehetne felülírni az `AdvancedHero`ban, hogy a sebzés után xp-t is növeljen. És akkor nem kell duplikálni a `Battle` logikáját.